jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/log/RuntimeLogPresenterTest.java

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.workbench.pr.client.editors.instance.log;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.safehtml.client.SafeHtmlTemplates;
+import com.google.gwt.safehtml.shared.SafeHtml;
+import com.google.gwt.safehtml.shared.SafeHtmlUtils;
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.jboss.errai.common.client.api.Caller;
+import org.jbpm.workbench.common.client.util.DateUtils;
+import org.jbpm.workbench.pr.client.resources.i18n.Constants;
+import org.jbpm.workbench.pr.client.util.LogUtils;
+import org.jbpm.workbench.pr.model.RuntimeLogSummary;
+import org.jbpm.workbench.pr.service.ProcessRuntimeDataService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.ocpsoft.prettytime.PrettyTime;
+import org.uberfire.mocks.CallerMock;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.*;
+
+import static org.uberfire.client.views.pfly.widgets.Moment.Builder.moment;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class RuntimeLogPresenterTest {
+
+    @Mock
+    RuntimeLogPresenter.RuntimeLogView view;
+
+    Caller<ProcessRuntimeDataService> processRuntimeDataServiceCaller;
+
+    @Mock
+    ProcessRuntimeDataService processRuntimeDataService;
+
+    @InjectMocks
+    RuntimeLogPresenter presenter;
+
+    private String serverTemplateId = "serverTemplateId";
+    private String deploymentId = "deploymentId";
+    private String processName = "processName";
+    private String testTask = "testTask";
+    private Long processInstanceId = 1L;
+
+    private Date logDate = new Date();
+    private String prettyTime;
+    private String techTime;
+
+    @Before
+    public void setup() {
+        processRuntimeDataServiceCaller = new CallerMock<ProcessRuntimeDataService>(processRuntimeDataService);
+        presenter.setProcessRuntimeDataService(processRuntimeDataServiceCaller);
+        prettyTime = moment().milliseconds(logDate.getTime() - new Date().getTime()).fromNow();
```

Comment:
Calling moment API in the Java test probably won't return any value as it relies on JavaScript to work, so I'm not sure there is much value to do this call as prettyTime will always be an empty String.